pytorch_lightning/trainer/trainer.py

```diff
@@ -2479,6 +2480,33 @@ def _active_loop(self) -> Optional[Union[FitLoop, EvaluationLoop, PredictionLoop
     Logging properties
     """
 
+    @property
+    def logger(self) -> Optional[LightningLoggerBase]:
+        return self._logger
+
+    @logger.setter
+    def logger(self, new_logger: Optional[LightningLoggerBase]) -> None:
+        self._logger = new_logger
+        if not new_logger:
+            self._loggers = []
+        elif isinstance(new_logger, LoggerCollection):
+            self._loggers = list(new_logger)
+        else:
+            self._loggers = [new_logger]
+
+    @property
+    def loggers(self) -> List[LightningLoggerBase]:
+        return self._loggers
+
+    @loggers.setter
+    def loggers(self, new_loggers: Optional[Iterable[LightningLoggerBase]]) -> None:
+        if new_loggers:
+            self._loggers = list(new_loggers)
```

Comment:
```suggestion
            self._loggers = new_loggers
```

Reply:
This is here because we want to enforce that the iterable is always stored as a list. See https://github.com/PyTorchLightning/pytorch-lightning/issues/11232#issuecomment-1007150402 for more details. Basically, we could pass in a list, a dict, or a LoggerCollection, but we always want it to be stored as a list :)